### PR TITLE
feat(evolution): purchase_rejection sensor + runtime wiring (Task 1 of Path A)

### PR DESCRIPTION
**Pull request by laofahai**
## Summary

First concrete Sensor for Spec 55 Evolution System — Task 1 of Path A (end-to-end AI evolution demo). Delivers three layers of plumbing plus one real sensor:

1. **Sensor collection at capability startup**: `collectCapabilityDefinitions` now extracts `cap.extensions.sensors` into `CollectedDefinitions.sensors`
2. **Runtime factory**: New `createEvolutionRuntime({sensors, query, memoryStore?})` instantiates `SignalBus + MemoryEngine + InsightEngine + AwarenessEngine + EvolutionCycle` and registers every sensor on the bus
3. **Dev boot wiring**: `dev.ts` calls `createEvolutionRuntime` at startup and logs the registered count
4. **First sensor**: `purchase_rejection_pattern` in `cap-purchase-demo` — counts `reject_purchase_request` succeeded events in `execution_log` over a rolling 30-day window

Relates to #88 (Evolution system — Insight + Proposal cycle).

## Cross-model review (2 rounds)

**Round 1** — Codex caught two blockers in the initial commit:
- **P1 (fixed)**: `extensions.sensors` was dead config — nothing read it at startup. Fixed by adding collection in `collect-capabilities.ts` + `createEvolutionRuntime` factory + wiring in `dev.ts`.
- **P2 (fixed)**: Sensor counted current `status=rejected` rows, not rejection events. A rejected-then-resubmitted request has `status=pending` again, erasing the event. Fixed by querying `execution_log` for `action_name=reject_purchase_request` / `status=succeeded`, filtering on `completed_at`.

Gemini (round 1) flagged a minor — null-timestamp records being treated as "recent" could cause migration-era false spikes. Fixed: records with null/invalid `completed_at` are now **excluded**.

**Round 2** — Codex found two more real issues, deliberately deferred to a follow-up to keep this PR focused (see #179):
- execution_log queries in `dev.ts` go to business `devDataProvider`, but `execution_log` lives in `SystemDataProvider` / `ExecutionLogger` — so the sensor will read 0 rows in dev
- `evolutionRuntime` is built in `dev.ts` but never reaches `transportCtx`, so MCP/HTTP cannot trigger `runCycle()`

Both are planned for Task 4 of Path A (Evolution MCP tools), which naturally needs runtime exposure anyway.

## Wiring chain (end-to-end, verified)

1. `capability.ts` declares `extensions.sensors: [purchaseRejectionPattern]`
2. `CapabilityExtensions.sensors?: Sensor[]` in core types
3. `collectCapabilityDefinitions` accumulates into `CollectedDefinitions.sensors`
4. `dev.ts` calls `createEvolutionRuntime({sensors: collected.sensors, query})`
5. Factory calls `signalBus.registerSensor(sensor)` for each
6. `evolutionCycle.runCycle(ctx)` -> `signalBus.collectSignals(ctx)` -> each sensor's `detect()`

Integration test `packages/core/__tests__/life-system/evolution-runtime.test.ts` exercises steps 3-6 end-to-end.

## Test plan

- [x] `bun run typecheck` — clean
- [x] `bun run check` — clean (1 pre-existing biome schema-version info, unrelated)
- [x] `bun test addons/demo/cap-purchase-demo/__tests__/purchase-rejection-pattern.test.ts` — 6 pass / 24 expects
- [x] `bun test packages/core/__tests__/life-system/evolution-runtime.test.ts` — 3 pass / 9 expects (new integration test)
- [x] `bun test` full suite — 4193 pass / 3 skip / 0 fail
- [x] Cross-model review (codex x2, gemini x1) — all blockers fixed, remaining gaps tracked in #179

## Files changed

**New (2)**:
- `packages/core/src/life-system/runtime.ts` — `createEvolutionRuntime` factory (139 lines)
- `packages/core/__tests__/life-system/evolution-runtime.test.ts` — integration test (123 lines)
- `addons/demo/cap-purchase-demo/src/sensors/purchase-rejection-pattern.ts` — sensor (79 lines across two commits)
- `addons/demo/cap-purchase-demo/__tests__/purchase-rejection-pattern.test.ts` — sensor test (148 lines)

**Modified**:
- `packages/cli/src/commands/startup/collect-capabilities.ts` — collect sensors
- `packages/cli/src/commands/dev.ts` — create runtime at boot
- `packages/core/src/index.ts`, `life-system/index.ts`, `server-entry.ts` — exports
- `addons/demo/cap-purchase-demo/src/capability.ts` — register sensor
- `addons/demo/cap-purchase-demo/src/index.ts` — re-export

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a purchase rejection pattern sensor that reports recent rejected purchase counts, confidence, and windowed context.
  * Added an evolution runtime to register/run sensors and exposed runtime creation in the public API.
  * Development startup now collects sensors and logs the count of registered sensors.

* **Tests**
  * Added tests covering the sensor behavior, timestamp/windowing rules, runtime wiring, query precedence, and related edge cases.

### CODE REVIEW

File: packages/cli/src/commands/dev.ts [hunk 1]
```diff
@@ -154,6 +159,28 @@ export const devCommand = defineCommand({
...
+    // any caller of evolutionRuntime.evolutionCycle.runCycle() can see them.
+    // A periodic driver is intentionally NOT started here — triggering is left
+    // to explicit callers (MCP tools, admin UI, scheduled jobs).
+    // Adapt DataProvider.query (concrete row type) to SensorContext.query (generic T).
+    // The cast is sound because callers (sensors) choose T to match their expected shape.
+    const evolutionQuery: SensorContext["query"] = async <T>(
+      schema: string,
+      filter?: Record<string, unknown>,
+    ) => {
+      const rows = await devDataProvider.query(schema, filter ?? {});
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/cli/src/commands/dev.ts:173]: Deferred to #179 (tracked separately). Both of your top-2 findings — (a) `devDataProvider` not reaching `execution_log` and (b) `evolutionRuntime` not flowing into `transportCtx` — are genuinely correct and will be fixed in Task 4 of Path A, which is where the MCP tools need the runtime exposure anyway. Keeping this PR focused on the sensor + factory + registration plumbing; the end-to-end dev loop is the next PR.

File: packages/cli/src/commands/dev.ts [hunk 2]
```diff
@@ -154,6 +159,28 @@ export const devCommand = defineCommand({
...
+      return rows as T[];
+    };
+    const evolutionRuntime = createEvolutionRuntime({
+      sensors: collected.sensors,
+      query: evolutionQuery,
+    });
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/cli/src/commands/dev.ts:179]: Deferred to #179 — same scope decision as the previous comment. `evolutionRuntime` is intentionally built but not yet handed to transports in this PR; that wiring lands with the MCP tools (Task 4), which need to reach the runtime to call `runCycle()` / `list_insights()` anyway.

File: addons/demo/cap-purchase-demo/src/sensors/purchase-rejection-pattern.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    // not a legitimate rejection event. Excluding (rather than including) is
+    // the conservative choice: we under-report rather than spike the signal
+    // and trigger a false positive insight.
+    const recentRejections = records.filter((record) => {
+      const raw = record.completed_at;
+      if (raw == null) return false;
+      const ts = raw instanceof Date ? raw : new Date(raw);
+      if (Number.isNaN(ts.getTime())) return false;
+      return ts.getTime() >= windowStart.getTime();
+    });
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/demo/cap-purchase-demo/src/sensors/purchase-rejection-pattern.ts:82]: Accepted as tech debt — acknowledged in the PR body. The simple equality filter on `DataProvider.query` cannot express date ranges today, so we fetch + post-filter. When execution_log volume warrants it, this should push the predicate into the data layer (or into a dedicated `cap-memory-drizzle` query helper). Tracking informally under #88.

File: packages/core/src/life-system/runtime.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    },
+    get awarenessEngine() {
+      return innerCycle.awarenessEngine;
+    },
+    runCycle(ctx?: SensorContext) {
+      const merged: SensorContext = {
+        timestamp: ctx?.timestamp ?? new Date(),
+        tenantId: ctx?.tenantId,
+        query: ctx?.query ?? opts.query,
+      };
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/life-system/runtime.ts:152]: Fixed in commit dbf2916. Switched to spread `...(ctx ?? { timestamp: new Date() })` so any future `SensorContext` fields (trace IDs, user metadata, etc.) flow through automatically; `timestamp` and `query` retain their explicit override semantics.

File: addons/demo/cap-purchase-demo/src/sensors/purchase-rejection-pattern.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/demo/cap-purchase-demo/src/sensors/purchase-rejection-pattern.ts:82]: Fixed in commit 5258a82. Added `windowEnd = ctx.timestamp` and changed the filter to `ts >= windowStart && ts <= windowEnd`. Future `completed_at` values (clock skew / corruption) are now excluded. Added a regression test case.

File: packages/core/src/life-system/runtime.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  const innerCycle = createEvolutionCycle({
+    signalBus,
+    memoryEngine,
+    awareness,
+    insightPromotion: opts.insightPromotion,
+  });
+
+  for (const sensor of opts.sensors) {
+    signalBus.registerSensor(sensor);
+  }
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/life-system/runtime.ts:132]: Fixed in commit 5258a82. `createEvolutionRuntime` now tracks registered sensor names in a `Set` and throws a descriptive error on duplicates (rather than silently overwriting). Added a regression test.
